syntax = "proto2";

package com.vmware.concord.hlf.storage;

message HlfTransaction {
   optional int64 version = 1;
   optional int64 block_id = 2;
   optional int64 tx_index = 3;
   optional string chain_id = 4;
   optional string chaincode_id = 5;
   optional string chaincode_version = 6;
   optional string input = 7;
   optional int64 status = 8;
}

message HlfBlock {
   optional int64 version = 1;
   optional int64 number = 2;
   optional bytes hash = 3;
   optional bytes parent_hash = 4;
   repeated bytes transaction = 5;
   optional int64 timestamp = 6;
}

message HlfState {
   optional int64 version = 1;
   optional string state = 2;
}

